test/TestProjects/MgmtMockAndSample/tests/Generated/Samples/Sample_ManagedHsmCollection_CreateANewManagedHSMPoolOrUpdateAnExistingManagedHSMPool.cs

```diff
@@ -0,0 +1,101 @@
+    {
+        // Create a new managed HSM Pool or update an existing managed HSM Pool
+        [NUnit.Framework.Test]
+        [NUnit.Framework.Ignore("Only verifying that the sample builds")]
+        public async Task CreateOrUpdate()
+        {
+            // this example is just showing the usage of "ManagedHsms_CreateOrUpdate" operation, for the dependent resources, they will have to be created separately.
+
+            // authenticate your client
+            ArmClient client = new ArmClient(new DefaultAzureCredential());
+
+            // this example assumes you already have this ResourceGroupResource created on azure
+            // for more information of creating ResourceGroupResource, please refer to the document of ResourceGroupResource
+            ResourceIdentifier resourceGroupResourceId = ResourceGroupResource.CreateResourceIdentifier("00000000-0000-0000-0000-000000000000", "hsm-group");
+            ResourceGroupResource resourceGroupResource = client.GetResourceGroupResource(resourceGroupResourceId);
+
+            // get the collection of this ManagedHsmResource
+            MgmtMockAndSample.ManagedHsmCollection collection = resourceGroupResource.GetManagedHsms();
+
+            // invoke the operation
+            ArmOperation<MgmtMockAndSample.ManagedHsmResource> lro = await collection.CreateOrUpdateAsync(WaitUntil.Completed, "hsm1", new ManagedHsmData(new AzureLocation("westus"))
+            {
+                Properties = new ManagedHsmProperties()
+                {
+                    Settings = BinaryData.FromObjectAsJson(new
+                    {
+                        config1 = "value1",
+                        config2 = "8427",
+                        config3 = "false",
+                        config4 = new[] { "1", "2" },
+                        config5 = new
+                        {
+                            inner = "something"
+                        }
+                    }),
+                    ProtectedSettings = BinaryData.FromObjectAsJson(new
+                    {
+                        protected1 = "value2",
+                        protected2 = "10",
+                        protected3 = "false",
+                        protected4 = new[] { "1", "2", "3" },
+                        protected5 = new
+                        {
+                            protectedInner = "something else"
+                        }
+                    }),
+                    RawMessage = Convert.FromBase64String("PFX-or-PEM-blob"),
+                    TenantId = Guid.Parse("00000000-0000-0000-0000-000000000000"),
+                    InitialAdminObjectIds =
+{
+"00000000-0000-0000-0000-000000000000"
+},
+                    EnableSoftDelete = true,
+                    SoftDeleteRetentionInDays = 90,
+                    EnablePurgeProtection = true,
+                    NetworkAcls = new MhsmNetworkRuleSet()
+                    {
+                        VirtualNetworkRules =
+{
```

Comment:
Can we improve this layout?

Reply:
Unfortunately I did not find an approach to tweak the format in our generator